pkg/controller/appbindings.go

```diff
@@ -78,20 +80,23 @@ func (c *StashController) applyBackupAnnotationLogicForAppBinding(ab *appCatalog
 		inputs[apis.TargetName] = ab.Name
 		inputs[apis.TargetNamespace] = ab.Namespace
 		inputs[apis.TargetAppVersion] = ab.Spec.Version
+		inputs[apis.TargetAppGroup] = targetAppGroup
+		inputs[apis.TargetAppResource] = targetAppResource
+		inputs[apis.TargetAppType] = string(ab.Spec.Type)
 
 		err = resolve.ResolveBackupBlueprint(backupBlueprint, inputs)
 		if err != nil {
 			return err
 		}
 
 		// ensure Repository crd
-		err = c.ensureRepository(backupBlueprint, targetRef)
+		err = c.ensureRepository(backupBlueprint, targetRef, targetAppResource)
```

Comment:
Why are we using the `Resource` from AppBinding here?

Reply:
removed